### PR TITLE
[GREENHOST] Add profile fields

### PR DESCRIPTION
**Pull request by arsentur**
**Description:**
Add new fields to the user profile model.
Add them to the registration form and account settings form.

**Youtrack:**
https://youtrack.raccoongang.com/issue/GREENHOST-93

### CODE REVIEW

File: common/djangoapps/student/models.py [hunk 1]
```diff
@@ -280,13 +299,47 @@ class Meta(object):
...
     country = CountryField(blank=True, null=True)
+
+    REGION_AFRICA = 'africa'
+    REGION_ASIA = 'asia'
+    REGION_EUR = 'eur'
+    REGION_S_US = 's_us'
+    REGION_N_US = 'n_us'
+    REGION_MENA = 'mena'
+    REGION_UNDERSCORED = 'underscored'
+    REGION_CHOICES = (
```
Comment (AndreyLykhoman) [inline common/djangoapps/student/models.py]: Why do you duplicate choices in properties?  I think this is odd.

File: openedx/core/djangoapps/user_api/accounts/serializers.py [hunk 2]
```diff
@@ -187,6 +194,11 @@ def validate_name(self, new_name):
             )
         return new_name
 
+    @staticmethod
+    def validate_gender(value):
+        """Returns gender's default value if value is empty string."""
+        return value or UserProfile.GENDER_OTHER
```
Comment (AndreyLykhoman) [inline openedx/core/djangoapps/user_api/accounts/serializers.py]: I think We can use GENDER_DEFAULT property and delete other extra 'GENDER_*' properties. What do you think about it?

File: openedx/core/djangoapps/user_api/views.py [hunk 3]
```diff
@@ -901,6 +904,90 @@ def _add_terms_of_service_field(self, form_desc, required=True):
...
+        Keyword Arguments:
+            required (bool): Whether this field is required; defaults to True
+
+        """
+        region_label = _(u"Profession")
+        error_msg = _(u"Please select your profession.")
+
+        # The labels are marked for translation in UserProfile model definition.
+        options = [(name, _(label)) for name, label in
+                   UserProfile.PROFESSION_CHOICES]  # pylint: disable=translation-of-non-string
```
Comment (AndreyLykhoman) [inline openedx/core/djangoapps/user_api/views.py]: Why do you transfer to a new line?

File: openedx/core/djangoapps/user_api/views.py [hunk 4]
```diff
@@ -901,6 +904,90 @@ def _add_terms_of_service_field(self, form_desc, required=True):
...
+        Keyword Arguments:
+            required (bool): Whether this field is required; defaults to True
+
+        """
+        region_label = _(u"Region")
+        error_msg = _(u"Please select your Region.")
+
+        # The labels are marked for translation in UserProfile model definition.
+        options = [(name, _(label)) for name, label in
+                   UserProfile.REGION_CHOICES]  # pylint: disable=translation-of-non-string
```
Comment (AndreyLykhoman) [inline openedx/core/djangoapps/user_api/views.py]: Why do you transfer to a new line?

File: openedx/core/djangoapps/user_api/views.py [hunk 5]
```diff
@@ -901,6 +904,90 @@ def _add_terms_of_service_field(self, form_desc, required=True):
...
+        Keyword Arguments:
+            required (bool): Whether this field is required; defaults to True
+
+        """
+        region_label = _(u"Age")
+        error_msg = _(u"Please select your age.")
+
+        # The labels are marked for translation in UserProfile model definition.
+        options = [(name, _(label)) for name, label in
+                   UserProfile.AGE_CHOICES]  # pylint: disable=translation-of-non-string
```
Comment (AndreyLykhoman) [inline openedx/core/djangoapps/user_api/views.py]: Why do you transfer to a new line?

File: common/djangoapps/student/models.py [hunk 6]
```diff
@@ -253,14 +253,28 @@ class Meta(object):
...
     year_of_birth = models.IntegerField(blank=True, null=True, db_index=True)
+
+    CHOICE_UNDISCLOSED = 'undisclosed'
+
+    AGE_CHOICES = (
+        ('20', ugettext_noop('< 20')),
+        ('40', ugettext_noop('20-40')),
+        ('64', ugettext_noop('41-64')),
+        ('65', ugettext_noop('65+')),
+        ('undisclosed', ugettext_noop('Undisclosed'))
```
Comment (idegtiarov) [inline common/djangoapps/student/models.py]: If you add default variable you should use it as a key in the choices tuple.

File: common/djangoapps/student/models.py [hunk 7]
```diff
@@ -253,14 +253,28 @@ class Meta(object):
...
+
+    user_age = models.CharField(max_length=12, db_index=True, choices=AGE_CHOICES, default=CHOICE_UNDISCLOSED)
+
+    GENDER_OTHER = 'o'
     GENDER_CHOICES = (
         ('m', ugettext_noop('Male')),
         ('f', ugettext_noop('Female')),
         # Translators: 'Other' refers to the student's gender
-        ('o', ugettext_noop('Other/Prefer Not to Say'))
+        ('o', ugettext_noop('Other / Undisclosed'))
```
Comment (idegtiarov) [inline common/djangoapps/student/models.py]: ditto

File: common/djangoapps/student/models.py [hunk 8]
```diff
@@ -253,14 +253,28 @@ class Meta(object):
...
     )
     gender = models.CharField(
-        blank=True, null=True, max_length=6, db_index=True, choices=GENDER_CHOICES
+        max_length=6, db_index=True, default=GENDER_OTHER, choices=GENDER_CHOICES
```
Comment (idegtiarov) [inline common/djangoapps/student/models.py]: I would decrease max_length to 1

File: common/djangoapps/student/models.py [hunk 9]
```diff
@@ -280,13 +294,36 @@ class Meta(object):
         # Translators: 'Other' refers to the student's level of education
         ('other', ugettext_noop("Other education"))
     )
+
+    PROFESSION_CHOICES = (
+        ('media', ugettext_noop('Media professional, activist, human rights defender')),
+        ('other', ugettext_noop("Other")),
+        ('undisclosed', ugettext_noop("Undisclosed")),
```
Comment (idegtiarov) [inline common/djangoapps/student/models.py]: I would decrease the key for `undisclosed` choice to 5 chars and also decrease max_length

File: common/djangoapps/student/models.py [hunk 10]
```diff
@@ -280,13 +294,36 @@ class Meta(object):
...
     level_of_education = models.CharField(
         blank=True, null=True, max_length=6, db_index=True,
         choices=LEVEL_OF_EDUCATION_CHOICES
     )
     mailing_address = models.TextField(blank=True, null=True)
     city = models.TextField(blank=True, null=True)
     country = CountryField(blank=True, null=True)
+
+    REGION_CHOICES = (
+        ('africa', ugettext_noop("Africa")),
```
Comment (idegtiarov) [inline common/djangoapps/student/models.py]: Lets set 4 chars length keys for choices

File: common/djangoapps/student/models.py [hunk 11]
```diff
@@ -280,13 +294,36 @@ class Meta(object):
...
+        ('asia', ugettext_noop('Asia & Pacific')),
+        ('eur', ugettext_noop("Europe")),
+        ('s_us', ugettext_noop('Latin / South America')),
+        ('n_us', ugettext_noop("North America")),
+        ('mena', ugettext_noop("MENA")),
+        ('undisclosed', ugettext_noop("Undisclosed")),
+    )
+    region = models.CharField(
+        max_length=12, db_index=True, default=CHOICE_UNDISCLOSED, choices=REGION_CHOICES
```
Comment (idegtiarov) [inline common/djangoapps/student/models.py]: ditto

Comment (arsentur) [reply common/djangoapps/student/models.py]: Done

**Pull request merged**